internal/gcsx/multi_range_downloader_wrapper_test.go

```diff
@@ -402,3 +403,289 @@ func (t *mrdWrapperTest) Test_EnsureMultiRangeDownloader_ForceRecreateMRD() {
+}
+
+func (t *mrdWrapperCacheTest) Test_Cache_EvictionOnOverflow() {
+	// Arrange: Create 4 wrappers (cache max is 3)
+	wrappers := make([]*MultiRangeDownloaderWrapper, 4)
+	for i := range 4 {
+		obj := &gcs.MinObject{
+			Name:       fmt.Sprintf("file%d", i),
+			Size:       100,
+			Generation: int64(1000 + i),
+		}
+		wrapper, err := NewMultiRangeDownloaderWrapperWithClock(
+			t.mockBucket,
+			obj,
+			&clock.FakeClock{WaitTime: t.mrdTimeout},
+			&cfg.Config{},
+			t.cache,
+		)
+		assert.NoError(t.T(), err)
+		wrapper.Wrapped = fake.NewFakeMultiRangeDownloaderWithSleep(obj, t.objectData, time.Microsecond)
+		wrappers[i] = wrapper
+	}
+
+	// Act: Open and close all 4 wrappers (triggers eviction on 4th)
+	for i := range 4 {
+		wrappers[i].IncrementRefCount()
+		err := wrappers[i].DecrementRefCount()
+		assert.NoError(t.T(), err)
+	}
+
+	// Assert: First wrapper evicted (LRU), last 3 remain in cache
+	assert.Nil(t.T(), wrappers[0].Wrapped, "First wrapper's MRD should be closed (evicted)")
+	for i := range wrappers[1:] {
+		key := wrapperKey(wrappers[i+1])
+		assert.NotNil(t.T(), t.cache.LookUpWithoutChangingOrder(key), "Wrapper %d should be in cache", i+1)
+		assert.NotNil(t.T(), wrappers[i+1].Wrapped, "Wrapper %d MRD should exist (pooled)", i+1)
+	}
+}
+
+func (t *mrdWrapperCacheTest) Test_Cache_DeletedIfReopened() {
+	// Arrange: Create 3 wrappers and fill cache
+	wrappers := make([]*MultiRangeDownloaderWrapper, 3)
+	for i := range 3 {
+		obj := &gcs.MinObject{
+			Name:       fmt.Sprintf("file%d", i),
+			Size:       100,
+			Generation: int64(1000 + i),
+		}
+		wrapper, err := NewMultiRangeDownloaderWrapperWithClock(
+			t.mockBucket,
+			obj,
+			&clock.FakeClock{WaitTime: t.mrdTimeout},
+			&cfg.Config{},
+			t.cache,
+		)
+		assert.NoError(t.T(), err)
+		wrapper.Wrapped = fake.NewFakeMultiRangeDownloaderWithSleep(obj, t.objectData, time.Microsecond)
+		wrappers[i] = wrapper
+		wrappers[i].IncrementRefCount()
+		err = wrappers[i].DecrementRefCount()
+		assert.NoError(t.T(), err)
+	}
+
+	// Act: Reopen wrapper 0 and add 4th wrapper (triggers eviction)
+	wrappers[0].IncrementRefCount()
+
+	// Assert: wrapper 0 will be deleted from cache.
+	assert.Nil(t.T(), t.cache.LookUpWithoutChangingOrder(wrapperKey(wrappers[0])), "Wrapper 0 should not be in cache")
+}
+
+func (t *mrdWrapperCacheTest) Test_Cache_ConcurrentAddRemove() {
+	// Arrange
+	const numGoroutines = 10
+	const numIterations = 100
+	wg := sync.WaitGroup{}
+
+	// Act: Concurrent open/close cycles from multiple goroutines
+	for range numGoroutines {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for range numIterations {
+				t.mrdWrapper.IncrementRefCount()
+				err := t.mrdWrapper.DecrementRefCount()
+				assert.NoError(t.T(), err)
+			}
+		}()
+	}
+	wg.Wait()
+
+	// Assert: Final state is refCount=0 (no deadlocks or panics)
+	assert.Equal(t.T(), 0, t.mrdWrapper.refCount, "RefCount should be 0 after all operations")
```

Comment:
what are we testing wrt to cache in this test? can we check if the entry is present in cache?